shell/client/grainview.js

```diff
@@ -401,9 +410,9 @@ GrainView.prototype.sessionStatus = function () {
 GrainView.prototype.route = function () {
   this._dep.depend();
   if (this._token) {
-    return "/shared/" + this._token;
+    return "/shared/" + this._token + this._path;
```

Comment:
It seems clear that we do want `setPath()` to be able to change the query and fragment. I guess the question is: should they be passed as separate arguments, or should they be part of the single `path` argument? I think we want separate arguments, but in any case it's a new feature and I think it should be a separate pull request.